File: tests/counter.ts
import * as anchor from "@project-serum/anchor";
import { BN, Program, web3 } from '@project-serum/anchor'
import { Counter } from "../target/types/counter";
const assert = require('assert')

describe("counter", () => {
  // Configure the client to use the local cluster.
  const provider = anchor.AnchorProvider.env()
  anchor.setProvider(provider)

  const program = anchor.workspace.Counter as Program<Counter>;
  const owner = provider.wallet.publicKey;

  const counter = web3.PublicKey.findProgramAddressSync(
    [Buffer.from('Counter'), owner.toBuffer()],
    program.programId
  )[0]
  console.log("pass1");
  it('Inits the counter', async () => {
    try{
      await program.methods
        .initCounter()
        .accounts({
                    owner, 
                    counter
                    })
        .rpc()
      }catch(e){
        console.log("counter already initialized or error on it lol");
      }
  })
  it('Add 1 to the counter', async () => {
    const prevcounterAccount = await (await program.account.counter.fetch(counter)).count.toNumber();
    const add2 = await program.methods
      .add()
      .accounts({ owner, counter })
      .instruction()

    const tx = new web3.Transaction()
    tx.add(add2,add2,add2);
    await provider.sendAndConfirm(tx)
    const counterAccount = await program.account.counter.fetch(counter);
    console.log("so, your current count is: ", counterAccount.count.toNumber());
    assert.ok(counterAccount.count.toNumber() === prevcounterAccount+3); // I have added 3 times the call , because of that the 3
});
})

File: programs_py/counter.py
# counter
# Built with Seahorse v0.1.6

from seahorse.prelude import *

declare_id('DyTz2YqoanAHMhDqUM9UwfzakXHoGdueLogNnbLTkceL')

class Counter(Account):
  owner: Pubkey
  count: i64

@instruction
def init_counter(owner: Signer, counter: Empty[Counter]):
    counter = counter.init(payer =  owner, seeds = ['Counter', owner])
    counter.owner = owner.key()

@instruction
def add(owner: Signer, counter: Counter):
    assert owner.key() == counter.owner, 'This is not your counter!'

    counter.count +=1

